kuzzle-sdk.d.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Type definitions for kuzzle-sdk javascript client.
+ * 
+ * Project: Kuzzle Javascript SDK - https://github.com/kuzzleio/sdk-javascript
+ * Definitions by: Andréas 'ScreamZ' HANSS <andreas.hanss67@gmail.com>
+ * ---
+ * Github: https://github.com/ScreamZ
+ * LinkedIn: https://www.linkedin.com/in/andreas-hanss/
+ * 
+ * If you find any issues, feel free to report or make a pull request.
+ */
+declare module "kuzzle-sdk" {
+    export class Kuzzle {
+        constructor(protocol: EmbededProtocol | CustomProtocol, options: KuzzleOptions);
+
+        // Core
+        addListener(event: KuzzleEvent, handler: (event: any) => void): Kuzzle;
+        connect(): Promise<void>;
+        disconnect(): void;
+        flushQueue(): Kuzzle;
+        playQueue(): Kuzzle;
+        query(request: KuzzleRequest, options?: QueuableOpts): RawKuzzleResponse
+        startQueuing(): Kuzzle;
+        stopQueuing(): Kuzzle;
+
+        // Modules
+        bulk: Bulk
+        index: Index
+    }
+
+    type EmbededProtocol = "http" | "websocket" | "socketio"
+    type CustomProtocol = object
+
+    interface QueuableOpts {
+        queuable: boolean;
+    }
+
+    interface KuzzleOptions {

```

Comment:
`KuzzleOptions` have some default key used by Kuzzle SDK itself but a custom protocol could accept other key for its configuration

Reply:
In that case it's up to the user to extends this definition type the following:

```ts
interface MyUpdatedConfig extends KuzzleOptions {
    myCustomFieldAdd: boolean;
}

Reply:
Otherwise I can also allow any other key that way it's a bit more flexible but less « secure ». Hum… in fact this solution is not blocking the first I gave, experimented user could still create an interface.

```ts
interface SquareConfig {
     host: string;
     autoQueue?: boolean;
     // …
    [propName: string]: any;
}
```

Reply:
I'm ok with this solution :)